react/features/base/dialog/components/Dialog.web.js

```diff
@@ -41,28 +53,138 @@ class Dialog extends AbstractDialog {
     }
 
     /**
+     * Initializes a new Dialog instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this._onKeyDown = this._onKeyDown.bind(this);
+        this._setDialogElement = this._setDialogElement.bind(this);
+    }
+
+    /**
+     * React Component method that executes once component is mounted.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        this.updateButtonFocus();
+
+        // adds listener for enter key
+        document.addEventListener('keydown', this._onKeyDown);
+    }
+
+    /**
+     * Clears the listener.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this._onKeyDown);
+    }
+
+    /**
+     * React Component method that executes once component is updated.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidUpdate(prevProps) {
+        // if there is an update in any of the buttons enable/disable props
+        // update the focus if needed
+        if (prevProps.okDisabled !== this.props.okDisabled
+            || prevProps.cancelDisabled !== this.props.cancelDisabled) {
+            this.updateButtonFocus();
+        }
+    }
+
+    /**
+     * Updates focused button, if we have a reference to the dialog element
+     * focus on available button if there is no focus already.
+     *
+     * @returns {void}
+     */
+    updateButtonFocus() {
+        if (this._dialogElement) {
+
+            // if we have a focused element inside the dialog, skip changing
+            // the focus
+            if (this._dialogElement.contains(document.activeElement)) {
```

Comment:
So the whole logic in this method seems a little bit unclear for me. On this line we are checking if any element inside the dialog is focused and later we are choosing explicitly one of the buttons to be focused.

The goal is to focus the dialog or a button? And since we have this keydown handler, do we care what is focused really? Also what would happen if you focus the cancel button and press enter?